br-webui/views/camera.liquid

```diff
@@ -16,30 +38,70 @@
 	
 	const GST_PARAM_FILE = 'gstreamer2.param';
 	
-	function updateGst(reset) {
-		disableAll(true);
-	
-		var gstOptions;
-		
-		if (!reset) {
-			gstOptions = document.getElementById('gstOptions').value;
-		}
-		console.log("update gst");
-		socket.emit('update gstreamer', gstOptions);
-	}
-	
 	function loadOptions() {
-		socket.emit('load params', {'file':GST_PARAM_FILE});
-	}
+		$.get('/pipelines', function(data) {
+			console.log(`Load pipeline: ${JSON.stringify(data)}`)
+			const videoConfiguration = data.videos_configuration[0]			
+
+			// Detect stream type
+			const streamType = (function() {
+				for (const [streamType, content] of Object.entries(PIPELINES)) {
+					if (videoConfiguration.endpoint.includes(content.endpoint_prefix)) {
+						return streamType
+					}
+				}
+			})()
+			
+			// Get necessary values to configure pipeline
+			const values = (function() {
```

Comment:
what values exactly? can you think of a better name? having to read a 20 lines anonymous function to understand what are "values" is a little annoying

Reply:
Added a more explicit comment block.